src/hooks/useCheckAccessToken.ts

```diff
@@ -0,0 +1,27 @@
+import { useEffect, useState } from 'react';
+
+import { useNavigate } from 'react-router-dom';
+
+import useAccessToken from './useAccessToken';
+
+export default function useCheckAccessToken(): boolean {
+  const [ready, setReady] = useState(false);
+
+  const { accessToken, setAccessToken } = useAccessToken();
+
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    if (accessToken) {
+      setReady(true);
+    }
+  }, [accessToken, setAccessToken]);
+
+  useEffect(() => {
+    if (!accessToken) {
+      navigate('/');
+    }
+  }, [accessToken, navigate]);
+
+  return ready;
```

Comment:
p2: 인터페이스가 아래처럼 되면 좋을 것 같은데 어떠신가요?

```suggestion
  return { ready };
```